instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4/common/client/NettyConnectNetAttributesGetter.java

```diff
@@ -19,25 +19,42 @@
 final class NettyConnectNetAttributesGetter
     extends InetSocketAddressNetClientAttributesGetter<NettyConnectionRequest, Channel> {
 
+  @Override
+  public String transport(NettyConnectionRequest request, @Nullable Channel channel) {
+    return channel instanceof DatagramChannel ? IP_UDP : IP_TCP;
+  }
+
   @Nullable
   @Override
-  public InetSocketAddress getAddress(NettyConnectionRequest request, @Nullable Channel channel) {
-    SocketAddress remoteAddress = null;
-    if (channel != null) {
-      remoteAddress = channel.remoteAddress();
-    }
-    // remote address on end() may be null when connection hasn't been established
-    if (remoteAddress == null) {
-      remoteAddress = request.remoteAddressOnStart();
+  public String peerName(NettyConnectionRequest request, @Nullable Channel channel) {
+    SocketAddress requestedAddress = request.remoteAddressOnStart();
```

Comment:
it seems like all netty operations are at socket layer, can you clarify why this works for `net.peer.name`?

Reply:
My reasoning here was that in the establishing connection phase, the socket address passed at the start (before the DNS lookup & connection occurs) is actually a logical address